plugin_manager.py

```diff
@@ -59,11 +88,28 @@ def load_plugin(self, plugin_name):
             for index in range(len(self.loaded_plugins)):
                 if self.loaded_plugins[index]["priority"] < plugin_obj["priority"]:
                     self.loaded_plugins.insert(index, plugin_obj)
-                    return
+                    return 0
             self.loaded_plugins.append(plugin_obj)
-        else:
-            # plugin not found
-            pass
+            return 0
+        return 1
+
+    def load_filter(self, filter_name):
```

Comment:
This seems like quite a bit of duplication of code. Please add a TODO to figure out how we can fix this in the future.